### PR TITLE
PN-7453-componente-dialog-con-style-non-centralizzato

### PR DESCRIPTION
**Pull request by ImJimMoriarty**
## Short description
Replaced some `Dialog` components with `PnDialog`.

## List of changes proposed in this pull request
- replaced `DialogActions` with `PnDialogActions`
- replaced `DialogContent` with `PnDialogContent`
- replaced `Dialog` with `PnDialog`
- moved `Dialog`'s props of `PnDialog` at the bottom in order to override default props
- adjusted relative tests after changes

## How to test
Just `yarn start` inside PA's package and check if dialogs are correctly rendered.

### CODE REVIEW

**Review by AndreaCimini90 — COMMENTED** [review]
Excluding those dialog that are heavily customized, use the PnDialog

**Comment by ImJimMoriarty** [comment]
I recommend doing end-to-end tests to verify whether the styles of the dialogs reflect those of the originals.
I've performed various checks, and it seems to be working, but you can never be sure about small details...

File: packages/pn-personagiuridica-webapp/src/components/Deleghe/ConfirmationModal.tsx [hunk 1]
```diff
@@ -32,67 +31,69 @@ export default function ConfirmationModal({
   const isMobile = useIsMobile();
 
   return (
-    <Dialog
+    <PnDialog
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Deleghe/ConfirmationModal.tsx:34]: Try to rework the modal using all the dialog components such as PnDialogActions, PnDialogTitle and so on

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]
1. There are Dialogs that aren't managed yet. Search in all the code the Dialog and replace it, where it is possible, with PnDialog
2. In ApiKeys.page, we use `<Box sx={{ minWidth: isMobile ? '0' : '600px' }}>`. I should remove it and try to use the maxWidth props on Dialog component
3. In one modal we set sx={{}}, i should try to override the style without setting an empty object

File: packages/pn-commons/src/components/CodeModal/CodeModal.tsx [hunk 2]
```diff
@@ -5,9 +5,6 @@ import {
   AlertTitle,
```
[the remark this answers is not in the document]
Comment (ImJimMoriarty) [reply packages/pn-commons/src/components/CodeModal/CodeModal.tsx:5]: Removed `mt` from the button to eliminate excess margin.

File: packages/pn-pa-webapp/src/components/Notifications/ConfirmCancellationDialog.tsx [hunk 3]
```diff
@@ -4,14 +4,12 @@ import { useTranslation } from 'react-i18next';
 import {
```
[the remark this answers is not in the document]
Comment (ImJimMoriarty) [reply packages/pn-pa-webapp/src/components/Notifications/ConfirmCancellationDialog.tsx:4]: Resolved by making adjustments to the (sx) properties.

File: packages/pn-pa-webapp/src/components/Notifications/ConfirmCancellationDialog.tsx [hunk 4]
```diff
@@ -36,11 +34,9 @@ const ConfirmCancellationDialog: React.FC<Props> = ({ showModal, onClose, onConf
   }, [showModal]);
 
   return (
-    <Dialog open={showModal} data-testid="cancel-notification-modal">
-      <DialogTitle id="dialog-title" sx={{ p: 4 }}>
-        {t('detail.cancel-notification-modal.title')}
-      </DialogTitle>
-      <DialogContent sx={{ px: 4, pb: 4 }}>
+    <PnDialog open={showModal} data-testid="cancel-notification-modal">
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/components/Notifications/ConfirmCancellationDialog.tsx:37]: Checks the comments left on CodeModal about distance between Title and Content, and Content and Actions

File: packages/pn-personafisica-webapp/src/components/Contacts/CancelVerificationModal.tsx [hunk 5]
```diff
@@ -22,22 +23,24 @@ const CancelVerificationModal = ({ open, handleClose }: Props) => {
   };
 
   return (
-    <Dialog open={open} onClose={handleClose} data-testid="cancelVerificationModal">
-      <DialogTitle id="dialog-title" sx={{ pt: 4, px: 4 }}>
+    <PnDialog open={open} onClose={handleClose} data-testid="cancelVerificationModal">
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/CancelVerificationModal.tsx:26]: Checks the comments left on CodeModal about distance between Title and Content, and Content and Actions

File: packages/pn-personafisica-webapp/src/components/Contacts/DigitalContactElem.tsx [hunk 6]
```diff
@@ -71,52 +61,34 @@ const DeleteDialog: React.FC<DialogProps> = ({
...
+      <Button key="confirm" onClick={confirmHandler} variant="contained">
         {t('button.conferma')}
-      </Button>
-    </>
+      </Button>,
+    ]
   );
   return (
-    <Dialog
+    <PnDialog
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/DigitalContactElem.tsx:80]: Checks the comments left on CodeModal about distance between Title and Content, and Content and Actions

File: packages/pn-personafisica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx [hunk 7]
```diff
@@ -378,7 +372,7 @@ const DigitalContactsCodeVerificationProvider: FC<{ children?: ReactNode }> = ({
           errorMessage={errorMessage?.content}
         />
       )}
-      <Dialog
+      <PnDialog
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx:375]: Checks the comments left on CodeModal about distance between Title and Content, and Content and Actions

File: packages/pn-personafisica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx [hunk 8]
```diff
@@ -388,33 +382,35 @@ const DigitalContactsCodeVerificationProvider: FC<{ children?: ReactNode }> = ({
...
           <Button onClick={handleConfirm} variant="contained">
             {t('button.conferma')}
           </Button>
-        </DialogActions>
-      </Dialog>
-      <Dialog open={pecValidationOpen} data-testid="validationDialog">
-        <DialogTitle id="dialog-title" sx={{ pt: 4, px: 4 }}>
+        </PnDialogActions>
+      </PnDialog>
+      <PnDialog open={pecValidationOpen} data-testid="validationDialog">
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx:399]: Checks the comments left on CodeModal about distance between Title and Content, and Content and Actions

File: packages/pn-personagiuridica-webapp/src/components/Contacts/CancelVerificationModal.tsx [hunk 9]
```diff
@@ -22,22 +23,24 @@ const CancelVerificationModal = ({ open, handleClose }: Props) => {
   };
 
   return (
-    <Dialog open={open} onClose={handleClose} data-testid="cancelVerificationModal">
-      <DialogTitle id="dialog-title" sx={{ pt: 4, px: 4 }}>
+    <PnDialog open={open} onClose={handleClose} data-testid="cancelVerificationModal">
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/CancelVerificationModal.tsx:26]: Checks the comments left on CodeModal about distance between Title and Content, and Content and Actions

File: packages/pn-personagiuridica-webapp/src/components/Contacts/DigitalContactElem.tsx [hunk 10]
```diff
@@ -74,52 +64,34 @@ const DeleteDialog: React.FC<DialogProps> = ({
...
+      <Button key="confirm" onClick={confirmHandler} variant="contained">
         {t('button.conferma')}
-      </Button>
-    </>
+      </Button>,
+    ]
   );
   return (
-    <Dialog
+    <PnDialog
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/DigitalContactElem.tsx:83]: Checks the comments left on CodeModal about distance between Title and Content, and Content and Actions

File: packages/pn-personagiuridica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx [hunk 11]
```diff
@@ -325,7 +319,7 @@ const DigitalContactsCodeVerificationProvider: FC<{ children?: ReactNode }> = ({
           errorMessage={errorMessage?.content}
         />
       )}
-      <Dialog
+      <PnDialog
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx:322]: Checks the comments left on CodeModal about distance between Title and Content, and Content and Actions

File: packages/pn-personagiuridica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx [hunk 12]
```diff
@@ -335,33 +329,35 @@ const DigitalContactsCodeVerificationProvider: FC<{ children?: ReactNode }> = ({
...
           <Button onClick={handleConfirm} variant="contained">
             {t('button.conferma')}
           </Button>
-        </DialogActions>
-      </Dialog>
-      <Dialog open={pecValidationOpen} data-testid="validationDialog">
-        <DialogTitle id="dialog-title" sx={{ pt: 4, px: 4 }}>
+        </PnDialogActions>
+      </PnDialog>
+      <PnDialog open={pecValidationOpen} data-testid="validationDialog">
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Contacts/DigitalContactsCodeVerification.context.tsx:346]: Checks the comments left on CodeModal about distance between Title and Content, and Content and Actions

File: packages/pn-personagiuridica-webapp/src/components/Deleghe/AcceptDelegationModal.tsx [hunk 13]
```diff
@@ -143,18 +140,18 @@ const AcceptDelegationModal: React.FC<Props> = ({
     );
   }
   return (
-    <Dialog
+    <PnDialog
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Deleghe/AcceptDelegationModal.tsx:143]: Checks the comments left on CodeModal about distance between Title and Content, and Content and Actions

File: packages/pn-personagiuridica-webapp/src/components/Deleghe/AcceptDelegationModal.tsx [hunk 14]
```diff
@@ -143,18 +140,18 @@ const AcceptDelegationModal: React.FC<Props> = ({
...
+      <DialogTitle id="dialog-title">
         {isEditMode ? t('deleghe.edit-groups-title') : t('deleghe.associate-groups-title')}
       </DialogTitle>
-      <DialogContent>
-        <DialogContentText id="dialog-description" sx={{ textAlign: textPosition }}>
+      <PnDialogContent>
+        <DialogContentText id="dialog-description">
           {t('deleghe.associate-groups-subtitle')}
         </DialogContentText>
         <Divider sx={{ margin: '20px 0' }} />
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Deleghe/AcceptDelegationModal.tsx:157]: Margin must be 16px. Use my 2

File: packages/pn-personagiuridica-webapp/src/components/Deleghe/AcceptDelegationModal.tsx [hunk 15]
```diff
@@ -215,18 +212,11 @@ const AcceptDelegationModal: React.FC<Props> = ({
             sx={{ my: 2 }}
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Deleghe/AcceptDelegationModal.tsx:212]: Maybe this must be changed with mt:2

File: packages/pn-personagiuridica-webapp/src/components/Deleghe/ConfirmationModal.tsx [hunk 16]
```diff
@@ -25,74 +22,40 @@ export default function ConfirmationModal({
...
-  minHeight = '4em',
 }: Props) {
-  const theme = useTheme();
-  const fullScreen = useMediaQuery(theme.breakpoints.down('xs'));
-  const isMobile = useIsMobile();
-
   return (
-    <Dialog
-      fullScreen={fullScreen}
+    <PnDialog
```
Comment (AndreaCimini90) [inline packages/pn-personagiuridica-webapp/src/components/Deleghe/ConfirmationModal.tsx:27]: Checks the comments left on CodeModal about distance between Title and Content, and Content and Actions

**Review by AndreaCimini90 — DISMISSED** [review]

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**